File: reverse_cards/reverse_cards.rb
module ReverseCards
  def self.start
    cards = Array.new(100, 'back')

    (2..100).each do |n|
      i = n - 1
      while i < cards.size
        cards[i] = toggle_card(cards[i])
        i += n
      end
    end
    p print_back_cards(cards)
  end

  def self.toggle_card(card_status)
    card_status == 'front' ? 'back' : 'front'
  end

  def self.print_back_cards(cards)
    back_indexes = []
    cards.each_with_index do |card, ind|
      back_indexes << ind + 1 if card == 'back'
    end
    back_indexes
  end
end

ReverseCards.start

File: reverse_cards/reverse_cards.js
var ReverseCards = {
    cards: [],
    initCards: function() {
        var cards = new Array(100);
        for (var i = 0; i < cards.length ; i++) {
            cards[i] = 'back';
        }
        this.cards = cards;
    },
    start: function() {
        this.initCards();
        for (var i = 2; i <= 100; i++) {
            var j = i - 1;
            while (j < this.cards.length) {
                this.cards[j] = this.toggleCard(this.cards[j]);
                j += i;
            }
        }
        var k = 0;
        while (k < this.cards.length) {
            if (this.cards[k] == 'back') {
                console.log(k + 1);
            }
            k++;
        }
    },
    toggleCard: function(card) {
        if (card == 'back') {
            return 'front';
        } else {
            return 'back';
        }
    }
};

ReverseCards.start();
